### ISSUE TITLE
Crash running "rake spec:device" in template app on Android 5.0 CM : java.lang.UnsatisfiedLinkError: JNI_ERR returned from JNI_OnLoad

### ISSUE THREAD

**Issue by buzzware** [HipByte/RubyMotion#194]
The error is java.lang.UnsatisfiedLinkError: JNI_ERR returned from JNI_OnLoad in "/data/app/com.yourcompany.blah-2/lib/arm/libpayload.so". 

See log below. "rake device" runs app fine.

garymac:rubymotion gary$ adb devices
List of devices attached 
HT29TW311667	device

garymac:rubymotion gary$ motion create --template=android blah
    Create blah
    Create blah/.gitignore
    Create blah/app/main_activity.rb
    Create blah/Gemfile
    Create blah/Rakefile
    Create blah/spec/main_spec.rb
garymac:rubymotion gary$ cd blah
garymac:blah gary$ rake device --trace
** Invoke device (first_time)
** Invoke build (first_time)
** Execute build
mkdir -p ./build/Development-21
mkdir -p ./build/Development-21/java/com/yourcompany/blah
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" package -f -M "./build/Development-21/AndroidManifest.xml" -A "./assets" -S "./resources" -I "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -m -J "./build/Development-21/java"  --auto-add-overlay
mkdir -p ./build/Development-21/classes
RUBYOPT='' /Library/RubyMotion/bin/android/gen_bridge_metadata  -o "./build/Development-21/R.bridgesupport" 
VM_PLATFORM=android VM_KERNEL_PATH="/Library/RubyMotion/data/android/21/armv5te/kernel-armv5te.bc" arch -i386 "/Library/RubyMotion/bin/ruby" --uses-bs "./build/Development-21/R.bridgesupport" --uses-bs "/Library/RubyMotion/data/android/21/BridgeSupport/android.bridgesupport" --emit-llvm "./build/Development-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o.bc" MREP_9790262bba2dd1e7151ad103d72267e39b38faad "./app/main_activity.rb"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -c "./build/Development-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o.bc" -o "./build/Development-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o"
mkdir -p ./build/Development-21/jni
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -mtune=xscale -MMD -MP -fpic -ffunction-sections -funwind-tables -fexceptions -fstack-protector -fno-rtti -fno-strict-aliasing -O0 -g3 -fno-omit-frame-pointer -DANDROID -I"/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm/usr/include" -Wformat -Werror=format-security -c "./build/Development-21/jni/payload.cpp" -o "./build/Development-21/jni/payload.o"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang++ -Wl,-soname,libpayload.so -shared --sysroot="/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm" -lgcc  -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -no-canonical-prefixes -target armv5te-none-linux-androideabi  -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -O0 -g3 "./build/Development-21/jni/payload.o" "./build/Development-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o" -o "./build/Development-21/lib/armeabi/libpayload.so" -L"/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm/usr/lib" -lstdc++ -lc -lm -llog -L"/Library/RubyMotion/data/android/21/armv5te" -lrubymotion-static -L/Users/gary/Applications/dev/android-rubymotion/ndk/sources/cxx-stl/stlport/libs/armeabi -lstlport_static
ln -s lib libs
/usr/bin/install -p /Users/gary/Applications/dev/android-rubymotion/ndk/prebuilt/android-arm/gdbserver/gdbserver ./build/Development-21/lib/armeabi
mkdir -p ./build/Development-21/classes
/usr/bin/javac -d "./build/Development-21/classes" -classpath "./build/Development-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Development-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Development-21/java/com/yourcompany/blah/MainActivity.java"
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/dx" --dex --output "./build/Development-21/classes.dex" "./build/Development-21/classes" "/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar" /Library/RubyMotion/data/android/21/rubymotion.jar
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" package -f -M "./build/Development-21/AndroidManifest.xml" -A "./assets" -S "./resources" -I "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -F "./build/Development-21/blah.apk" --auto-add-overlay
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "classes.dex"
 'classes.dex'...
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "lib/armeabi/libpayload.so"
 'lib/armeabi/libpayload.so'...
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "lib/armeabi/gdbserver"
 'lib/armeabi/gdbserver'...
/usr/bin/jarsigner -digestalg SHA1 -storepass android -keystore "/Users/gary/.android/debug.keystore" "./build/Development-21/blah.apk" androiddebugkey
jar signed.

Warning: 
No -tsa or -tsacert is provided and this jar is not timestamped. Without a timestamp, users may not be able to validate this jar after the signer certificate's expiration date (2044-03-01) or after any future revocation date.
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/zipalign" -f 4 "./build/Development-21/blah.apk" "./build/Development-21/blah.apk-aligned"
/bin/mv "./build/Development-21/blah.apk-aligned" "./build/Development-21/blah.apk"
** Invoke device:install (first_time)
** Execute device:install
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d install -r "./build/Development-21/blah.apk"
4951 KB/s (858722 bytes in 0.169s)
	pkg: /data/local/tmp/blah.apk
Success
** Invoke device:start (first_time)
** Execute device:start
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d logcat -c
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d shell am start -a android.intent.action.MAIN -n com.yourcompany.blah/.MainActivity
Starting: Intent { act=android.intent.action.MAIN cmp=com.yourcompany.blah/.MainActivity }
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d forward tcp:33333 tcp:33333
"/Library/RubyMotion/bin/android/repl" "/Library/RubyMotion/data/android/21/armv5te/kernel-armv5te.bc" 0.0.0.0 33333
--------- beginning of main
--------- beginning of system
>> quit
=> false
>> exit
Remote process has quit.
** Execute device
garymac:blah gary$ rake spec:device --trace
** Invoke spec:device (first_time)
** Execute spec:device
** Invoke device (first_time)
** Invoke build (first_time)
** Execute build
mkdir -p ./build/Testing-21
mkdir -p ./build/Testing-21/java/com/yourcompany/blah
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" package -f -M "./build/Testing-21/AndroidManifest.xml" -A "./assets" -S "./resources" -I "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -m -J "./build/Testing-21/java"  --auto-add-overlay
mkdir -p ./build/Testing-21/classes
RUBYOPT='' /Library/RubyMotion/bin/android/gen_bridge_metadata  -o "./build/Testing-21/R.bridgesupport" 
VM_PLATFORM=android VM_KERNEL_PATH="/Library/RubyMotion/data/android/21/armv5te/kernel-armv5te.bc" arch -i386 "/Library/RubyMotion/bin/ruby" --uses-bs "./build/Testing-21/R.bridgesupport" --uses-bs "/Library/RubyMotion/data/android/21/BridgeSupport/android.bridgesupport" --emit-llvm "./build/Testing-21/obj/local/armeabi/Library/RubyMotion/lib/motion/spec.rb.o.bc" MREP_fdff6f5bd64f3d02aa0c8e774f7c6ec27d6f9489 "/Library/RubyMotion/lib/motion/spec.rb"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -c "./build/Testing-21/obj/local/armeabi/Library/RubyMotion/lib/motion/spec.rb.o.bc" -o "./build/Testing-21/obj/local/armeabi/Library/RubyMotion/lib/motion/spec.rb.o"
VM_PLATFORM=android VM_KERNEL_PATH="/Library/RubyMotion/data/android/21/armv5te/kernel-armv5te.bc" arch -i386 "/Library/RubyMotion/bin/ruby" --uses-bs "./build/Testing-21/R.bridgesupport" --uses-bs "/Library/RubyMotion/data/android/21/BridgeSupport/android.bridgesupport" --emit-llvm "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/spec/main_spec.rb.o.bc" MREP_4aa211f4be82b2a66498f0d92ae877b46155e725 "./spec/main_spec.rb"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -c "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/spec/main_spec.rb.o.bc" -o "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/spec/main_spec.rb.o"
VM_PLATFORM=android VM_KERNEL_PATH="/Library/RubyMotion/data/android/21/armv5te/kernel-armv5te.bc" arch -i386 "/Library/RubyMotion/bin/ruby" --uses-bs "./build/Testing-21/R.bridgesupport" --uses-bs "/Library/RubyMotion/data/android/21/BridgeSupport/android.bridgesupport" --emit-llvm "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o.bc" MREP_64c92dcfa8bab6a42c6867f5f9e20f6ad303c656 "./app/main_activity.rb"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -c "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o.bc" -o "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o"
mkdir -p ./build/Testing-21/jni
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang -no-canonical-prefixes -target armv5te-none-linux-androideabi -march=armv5te -mthumb -msoft-float -marm -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -mtune=xscale -MMD -MP -fpic -ffunction-sections -funwind-tables -fexceptions -fstack-protector -fno-rtti -fno-strict-aliasing -O0 -g3 -fno-omit-frame-pointer -DANDROID -I"/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm/usr/include" -Wformat -Werror=format-security -c "./build/Testing-21/jni/payload.cpp" -o "./build/Testing-21/jni/payload.o"
/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/llvm-3.4/prebuilt/darwin-x86_64/bin/clang++ -Wl,-soname,libpayload.so -shared --sysroot="/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm" -lgcc  -gcc-toolchain "/Users/gary/Applications/dev/android-rubymotion/ndk/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64" -no-canonical-prefixes -target armv5te-none-linux-androideabi  -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -O0 -g3 "./build/Testing-21/jni/payload.o" "./build/Testing-21/obj/local/armeabi/Library/RubyMotion/lib/motion/spec.rb.o" "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/spec/main_spec.rb.o" "./build/Testing-21/obj/local/armeabi/Users/gary/repos/rubymotion/blah/app/main_activity.rb.o" -o "./build/Testing-21/lib/armeabi/libpayload.so" -L"/Users/gary/Applications/dev/android-rubymotion/ndk/platforms/android-21/arch-arm/usr/lib" -lstdc++ -lc -lm -llog -L"/Library/RubyMotion/data/android/21/armv5te" -lrubymotion-static -L/Users/gary/Applications/dev/android-rubymotion/ndk/sources/cxx-stl/stlport/libs/armeabi -lstlport_static
ln -s lib libs
/usr/bin/install -p /Users/gary/Applications/dev/android-rubymotion/ndk/prebuilt/android-arm/gdbserver/gdbserver ./build/Testing-21/lib/armeabi
mkdir -p ./build/Testing-21/classes
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/Context.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/Error.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/FalseClass.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/MainActivity.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/Numeric.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/Should.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/Specification.java"
/usr/bin/javac -d "./build/Testing-21/classes" -classpath "./build/Testing-21/classes":"/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar":"/Library/RubyMotion/data/android/21/rubymotion.jar" -sourcepath "./build/Testing-21/java" -target 1.5 -bootclasspath "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -encoding UTF-8 -g -source 1.5 "./build/Testing-21/java/com/yourcompany/blah/TrueClass.java"
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/dx" --dex --output "./build/Testing-21/classes.dex" "./build/Testing-21/classes" "/Users/gary/Applications/dev/android-rubymotion/sdk/tools/support/annotations.jar" /Library/RubyMotion/data/android/21/rubymotion.jar
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" package -f -M "./build/Testing-21/AndroidManifest.xml" -A "./assets" -S "./resources" -I "/Users/gary/Applications/dev/android-rubymotion/sdk/platforms/android-21/android.jar" -F "./build/Testing-21/blah.apk" --auto-add-overlay
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "classes.dex"
 'classes.dex'...
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "lib/armeabi/libpayload.so"
 'lib/armeabi/libpayload.so'...
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/aapt" add -f "blah.apk" "lib/armeabi/gdbserver"
 'lib/armeabi/gdbserver'...
/usr/bin/jarsigner -digestalg SHA1 -storepass android -keystore "/Users/gary/.android/debug.keystore" "./build/Testing-21/blah.apk" androiddebugkey
jar signed.

Warning: 
No -tsa or -tsacert is provided and this jar is not timestamped. Without a timestamp, users may not be able to validate this jar after the signer certificate's expiration date (2044-03-01) or after any future revocation date.
"/Users/gary/Applications/dev/android-rubymotion/sdk/build-tools/21.1.2/zipalign" -f 4 "./build/Testing-21/blah.apk" "./build/Testing-21/blah.apk-aligned"
/bin/mv "./build/Testing-21/blah.apk-aligned" "./build/Testing-21/blah.apk"
** Invoke device:install (first_time)
** Execute device:install
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d install -r "./build/Testing-21/blah.apk"
4354 KB/s (1132890 bytes in 0.254s)
	pkg: /data/local/tmp/blah.apk
Success
** Invoke device:start (first_time)
** Execute device:start
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d logcat -c
"/Users/gary/Applications/dev/android-rubymotion/sdk/platform-tools/adb" -d shell am start -a android.intent.action.MAIN -n com.yourcompany.blah/.MainActivity
Starting: Intent { act=android.intent.action.MAIN cmp=com.yourcompany.blah/.MainActivity }
--------- beginning of main
--------- beginning of system
E/com/yourcompany/blah(11186): Uncaught exception when initializing `spec.rb.o' scope -- aborting
--------- beginning of crash
E/AndroidRuntime(11186): FATAL EXCEPTION: main
E/AndroidRuntime(11186): Process: com.yourcompany.blah, PID: 11186
E/AndroidRuntime(11186): java.lang.UnsatisfiedLinkError: JNI_ERR returned from JNI_OnLoad in "/data/app/com.yourcompany.blah-2/lib/arm/libpayload.so"
E/AndroidRuntime(11186): 	at java.lang.Runtime.loadLibrary(Runtime.java:371)
E/AndroidRuntime(11186): 	at java.lang.System.loadLibrary(System.java:989)
E/AndroidRuntime(11186): 	at com.yourcompany.blah.MainActivity.<clinit>(MainActivity.java:10)
E/AndroidRuntime(11186): 	at java.lang.reflect.Constructor.newInstance(Native Method)
E/AndroidRuntime(11186): 	at java.lang.Class.newInstance(Class.java:1572)
E/AndroidRuntime(11186): 	at android.app.Instrumentation.newActivity(Instrumentation.java:1065)
E/AndroidRuntime(11186): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2230)
E/AndroidRuntime(11186): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2391)
E/AndroidRuntime(11186): 	at android.app.ActivityThread.access$900(ActivityThread.java:147)
E/AndroidRuntime(11186): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
E/AndroidRuntime(11186): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(11186): 	at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime(11186): 	at android.app.ActivityThread.main(ActivityThread.java:5256)
E/AndroidRuntime(11186): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11186): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(11186): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
E/AndroidRuntime(11186): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
** Execute device
garymac:blah gary$

**Comment by buzzware** [comment]
I can't get past this. I've tried this on Lollipop and 4.4.3 phones. I've tried Java 1.6, 1.7 and 1.8. 
The spec runner just doesn't seem to work at all now, although I have seen it work on occasion (that is, it launched the app without error, but didn't quit or report on any specs executed)

Java is configured as :

$ ls -la /System/Library/Frameworks/JavaVM.framework/Versions
total 64
drwxr-xr-x  11 root  wheel  374 10 Mar 10:56 .
drwxr-xr-x  12 root  wheel  408 17 Dec 02:20 ..
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.4 -> CurrentJDK
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.4.2 -> CurrentJDK
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.5 -> CurrentJDK
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.5.0 -> CurrentJDK
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.6 -> CurrentJDK
lrwxr-xr-x   1 root  wheel   10  9 Mar  2014 1.6.0 -> CurrentJDK
drwxr-xr-x   8 root  wheel  272 10 Mar 10:12 A
lrwxr-xr-x   1 root  wheel    1 10 Mar 09:41 Current -> A
lrwxr-xr-x   1 root  wheel   59 10 Mar 10:56 CurrentJDK -> /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents

garymac:Hello2 gary$ set | grep -i java
JAVA_HOME=/System/Library/Frameworks/JavaVM.framework/Versions/CurrentJDK/Home
JAVA_OPTS=-client
JAVA_VERSION=1.6
garymac:Hello2 gary$ cat /etc/launchd.conf | grep -i JAVA
setenv JAVA_VERSION 1.6
setenv JAVA_HOME /System/Library/Frameworks/JavaVM.framework/Versions/CurrentJDK/Home
garymac:Hello2 gary$ javac -version
javac 1.6.0_65
